.github/workflows/deploy-github-pages-release.yml

```
@@ -0,0 +1,298 @@
+          echo "${injection_code}" >> "${PATH_TO_POSYDON}/docs/conf.py"
+
+          # build the documentation
+          cd docs
+          make html
+          touch _build/html/.nojekyll
+          
+          # move and copy docs out of the POSYDON folder
+          tag_without_v=${tag//v}
+          mv _build/html/ $output_folder/$tag_without_v
+
+          # cleanup for the next build
+          make clean
+          cd ../
+        done
+
+    - name: Deploy to GitHub Pages
+      if: success()
+      uses: crazy-max/ghaction-github-pages@v2
+      with:
+        target_branch: gh-pages
+        build_dir: _build/
+        history: true
+        dry_run: true
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  
+  build_v2:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        fetch-tags: true
+        path: 'POSYDON'
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.11'
+  
+    - name: Get pip cache dir
+      id: pip-cache
+      run: |
+        python -m pip install --upgrade pip
+        echo "::set-output name=dir::$(pip cache dir)"
+
+    - name: Load pip cache
+      uses: actions/cache@v4
+      with:
+        path: ${{ steps.pip-cache.outputs.dir }}
+        key: ${{ runner.os }}-pip-${{ hashFiles('**/setup.py', '**/requirements.txt') }}
+        restore-keys: |
+          ${{ runner.os }}-pip-
+
+    - name: Install dependencies
+      run: |
+        # install pandoc for the documentation
+        sudo apt-get update -y
+        sudo apt-get install gfortran swig libhdf5-serial-dev libmpich-dev -y
+        sudo apt-get install pandoc -y
+
+        # install the dependencies in python
+        python -m pip install pandoc
+        python -m pip install coverage cpp-coveralls flake8 pytest
+
+    - name: Make Documentation
+      env:
+        PATH_TO_POSYDON: ./
+        PATH_TO_POSYDON_DATA: ./
+        MESA_DIR: ./
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        # make the output folder
+        output_folder=$GITHUB_WORKSPACE/_build/
+        echo "Output folder: $output_folder"
+        mkdir $output_folder
+
+        # move to the repository folder
+        export PATH_TO_POSYDON=$GITHUB_WORKSPACE/POSYDON
+        cd $PATH_TO_POSYDON
+
+        # REQUIRED FOR OLDER VERSIONS OF POSYDON BEFORE THE INCLUSION OF 
+        # THe _templates folder and versioning code.
+        # get the _templates folder and move it out of the POSYDON folder
+        cp -r $PATH_TO_POSYDON/docs/_source/_templates $GITHUB_WORKSPACE
+
+        # get the injection code from the release
+        injection_code=$(awk '/# INJECTION GRAB START/,/# INJECTION GRAB END/' "${PATH_TO_POSYDON}/docs/_source/conf.py")
+
+        # get the available tags
+        TAGS=$(git tag -l | sort -V | grep "v2" | grep -v "-dev")
```

```suggestion
        TAGS=$(git tag -l | sort -V | grep "v2" | grep -v "dev")
```
Using `"-dev"` throws an error for me when testing with `act`:

```
grep: invalid argument ‘ev’ for ‘--directories’
```
where grep interprets `"-dev"` as `"--directories ev"`. Using just `"dev"` works, as long as you think that's specific enough for the code's needs.

---

Yes, the same as I pointed out at the other place either you need the `-e` option of `grep` or leave out/mask the dash.